### PR TITLE
migrate to blockId

### PR DESCRIPTION
**Pull request by Aragar199**
The way we currently handle [block id](https://github.com/keep-starknet-strange/beerus/blob/56026b7579078a2ecb725e2a8d434b060bce6a50/examples/beerus-rpc/starknet_getEstimateFee.hurl#L7) is not compliant with the ecosystem([example](https://github.com/eqlabs/pathfinder/blob/1112b7532a998c5237df4dd8d76e96b7097c2011/doc/rpc/rpc_examples_v03.sh#L26))

Beerus needs to handle a BlockId similar to other R[PC Providers](https://github.com/starkware-libs/papyrus/blob/1ca9d93f5830b6e883e2ed9d4efc84fc2c8154e5/crates/papyrus_gateway/src/api.rs#L37)

# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing
- [ ] Other (please describe):

Issue Number: #351 

# Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: crates/beerus-rpc/src/api.rs [hunk 1]
```diff
@@ -85,16 +85,11 @@ pub trait BeerusApi {
     async fn get_transaction_by_hash(&self, tx_hash: &str) -> Result<Transaction, Error>;
 
     #[method(name = "getBlockTransactionCount")]
-    async fn get_block_transaction_count(
-        &self,
-        block_id_type: String,
-        block_id: String,
-    ) -> Result<u64, Error>;
+    async fn get_block_transaction_count(&self, block_id: String) -> Result<u64, Error>;
```
Comment (phklive) [inline crates/beerus-rpc/src/api.rs:88]: Valid for all other instances: Shouldn't we use BlockId as a type in the method definition params?

File: crates/beerus-rpc/src/lib.rs [hunk 2]
```diff
@@ -102,12 +101,9 @@ impl BeerusApiServer for BeerusRpc {
...
 
-    async fn get_block_transaction_count(
-        &self,
-        block_id_type: String,
-        block_id: String,
-    ) -> Result<u64, Error> {
-        let block_id = block_id_string_to_block_id_type(&block_id_type, &block_id).unwrap();
+    async fn get_block_transaction_count(&self, block_id: String) -> Result<u64, Error> {
+        let block_id: BlockId = serde_json::from_str(&block_id)
+            .map_err(|e| Error::Call(CallError::InvalidParams(anyhow::anyhow!(e.to_string()))))?;
```
Comment (phklive) [inline crates/beerus-rpc/src/lib.rs:106]: Valid for all other instances: 

Shouldn't we use BeerusApiError error handling here?

As follows: .map_err(|_| Error::from(BeerusApiError::InvalidCalldata))

As we are handling errors this way on other endpoints,

Please explain

**Review by phklive — COMMENTED** [review]
I have some questions that you could answer. 

Also the hurl files should be updated.

File: crates/beerus-rpc/src/lib.rs (hunk 2 above)
Comment (Aragar199) [reply crates/beerus-rpc/src/lib.rs:106]: Oh yeah, that's my bad. I'll change it to use BeerusApiError.

File: crates/beerus-rpc/src/api.rs [hunk 3]
```diff
@@ -121,7 +114,6 @@ pub trait BeerusApi {
     #[method(name = "getTransactionByBlockIdAndIndex")]
     async fn get_transaction_by_block_id_and_index(
         &self,
-        block_id_type: &str,
         block_id: &str,
```
[the remark this answers is not in the document]
Comment (drspacemn) [reply crates/beerus-rpc/src/api.rs:117]: They should all be `BlockId`

[the remark this answers is not in the document]
Comment (drspacemn) [reply crates/beerus-rpc/src/api.rs:117]: You can use papyrus RPC impl as a guide:
https://github.com/starkware-libs/papyrus/blob/96269d173405341dde449d2a030e9fa933d2467a/crates/papyrus_gateway/src/api.rs#L98

**Review by drspacemn — APPROVED** [review]
lgtm

File: examples/beerus-rpc/starknet_getBlockTransactionCount.hurl [hunk 4]
```diff
@@ -2,6 +2,6 @@ POST http://0.0.0.0:3030
 Content-Type: application/json
 {
     "jsonrpc":"2.0","method":"starknet_getBlockTransactionCount","params":[
-        "tag", "latest"
+        "latest"
```
Comment (phklive) [inline examples/beerus-rpc/starknet_getBlockTransactionCount.hurl:5]: Valid also for all other instances: Here you do not follow the same structure as in the other hurls for example: 
{
      "block_number": 22050
},

After testing it does work but is that normal ? 

Shouldn't it be: 

{
 "block_tag": "latest"
}

Or something in that range ?

**Review by phklive — COMMENTED** [review]
Only one comment, but lgtm, what do you think @drspacemn, repo looks clean now

File: examples/beerus-rpc/starknet_getBlockTransactionCount.hurl (hunk 4 above)
Comment (Aragar199) [reply examples/beerus-rpc/starknet_getBlockTransactionCount.hurl:5]: Screen Shot 2023-04-13 at 2 42 44 PM

That's the way it's done in the ecosystem

**Pull request merged**